CMakeLists.txt

```diff
@@ -27,21 +27,45 @@ target_link_libraries(${PROJECT_NAME}
 
 add_subdirectory(examples)
 
-install(DIRECTORY include/ DESTINATION include/)
-
+install(
+  DIRECTORY include/
+  DESTINATION include
+)
 install(
   TARGETS ${PROJECT_NAME}
-  EXPORT ${PROJECT_NAME}Targets
+  EXPORT export_${PROJECT_NAME}
   LIBRARY DESTINATION lib
   ARCHIVE DESTINATION lib
   RUNTIME DESTINATION bin
   INCLUDES DESTINATION include
 )
 
-ament_export_include_directories(include)
-ament_export_targets(${PROJECT_NAME}Targets)
-ament_export_dependencies(fmt)
-ament_export_dependencies(range-v3)
+# export targets
+set(export_targets ${export_targets};${PROJECT_NAME})
+export(EXPORT export_${PROJECT_NAME}
+  FILE "${PROJECT_BINARY_DIR}/export_${PROJECT_NAME}.cmake")
+
+# Create fpConfig.cmake
+set(CONF_INCLUDE_DIRS "${CMAKE_INSTALL_PREFIX}/include")
+configure_file(fpConfig.cmake.in
+  "${PROJECT_BINARY_DIR}/${CMAKE_FILES_DIRECTORY}/fpConfig.cmake" @ONLY)
+
+# Create fpConfigVersion.cmake
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(fpConfigVersion.cmake
+  VERSION ${PROJECT_VERSION}
+  COMPATIBILITY SameMajorVersion
+)
+
+# Install find_package cmake files
+install(FILES
+  "${PROJECT_BINARY_DIR}/${CMAKE_FILES_DIRECTORY}/fpConfig.cmake"
+  "${PROJECT_BINARY_DIR}/fpConfigVersion.cmake"
+  DESTINATION "share/${PROJECT_NAME}/cmake" COMPONENT dev)
+install(EXPORT export_${PROJECT_NAME}
```

Comment:
I'd recommend adding `NAMESPACE fp::` here so that when anyone includes your library, they can link to `fp::fp` instead of `fp`. It's CMake convention and recommended practice to namespace all exported targets because CMake can know with certainly that `fp::fp` is referring to a CMake target and not something like `libfp.a` because of the double semicolons in the name. Likewise, add
```cpp
add_library(fp::fp ALIAS fp)
```
higher up so that anyone directly using this CMakeLists.txt can also link to `fp::fp`.

Reply:
I didn't have this before and was linking to this with `fp::fp`. I can add these changes though.